pkg/syncer/metadatasyncer.go

```diff
@@ -789,6 +834,91 @@ func addLabelsToTopologyVCMap(ctx context.Context, nodeTopoObj csinodetopologyv1
 	}
 }
 
+// cnsvolumeoperationrequestCRUpdated checks if the cnsvolumeoperationrequest instance reserved field is updated
+func cnsvolumeoperationrequestCRUpdated(oldObj interface{}, newObj interface{}) {
+	ctx, log := logger.GetNewContextWithLogger()
+	// Verify both objects received.
+	var (
+		oldcnsvolumeoperationrequestObj cnsvolumeoperationrequestv1alpha1.CnsVolumeOperationRequest
+		newcnsvolumeoperationrequestObj cnsvolumeoperationrequestv1alpha1.CnsVolumeOperationRequest
+	)
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(
+		newObj.(*unstructured.Unstructured).Object, &newcnsvolumeoperationrequestObj)
+	if err != nil {
+		log.Errorf("cnsvolumeoperationrequestCRUpdated: failed to cast new object %+v to %s. Error: %+v", newObj,
+			cnsvolumeoperationrequest.CRDSingular, err)
+		return
+	}
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(
+		oldObj.(*unstructured.Unstructured).Object, &oldcnsvolumeoperationrequestObj)
+	if err != nil {
+		log.Errorf("cnsvolumeoperationrequestCRUpdated: failed to cast old object %+v to %s. Error: %+v", oldObj,
+			csinodetopology.CRDSingular, err)
+		return
+	}
+	if oldcnsvolumeoperationrequestObj.Status.Reserved != nil &&
+		newcnsvolumeoperationrequestObj.Status.Reserved != nil {
+		if oldcnsvolumeoperationrequestObj.Status.Reserved.Value() !=
+			newcnsvolumeoperationrequestObj.Status.Reserved.Value() {
+			// Update StoragePolicyUsage reserved field
+			restConfig, err := config.GetConfig()
+			if err != nil {
+				log.Errorf("failed to get Kubernetes config. Err: %+v", err)
+				return
+			}
+
+			cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig, cnsoperatorv1alpha1.GroupName)
+			if err != nil {
+				log.Errorf("Failed to create CnsOperator client. Err: %+v", err)
+				return
+			}
+			storagePolicyUsageList := &storagepolicyusagev1alpha1.StoragePolicyUsageList{}
+			err = cnsOperatorClient.List(ctx, storagePolicyUsageList, client.InNamespace(
+				newcnsvolumeoperationrequestObj.Namespace))
+			if err != nil {
+				log.Errorf("failed to list %s CR from supervisor namespace %q. Error: %+v",
+					storagepolicyusagev1alpha1.CRDSingular, newcnsvolumeoperationrequestObj.Namespace, err)
+				return
+			}
+			foundStoragePolicyUsageCR := false
+			storagePolicyUsageCR := storagepolicyusagev1alpha1.StoragePolicyUsage{}
+			if len(storagePolicyUsageList.Items) > 0 {
+				for _, item := range storagePolicyUsageList.Items {
+					if item.Spec.StoragePolicyId == newcnsvolumeoperationrequestObj.Status.StoragePolicyId {
```

Comment:
there could be multiple StoragePolicyUsage CRs for same policy id. we need to find one with given storage policy id, storage class name, resource type, resource group

Reply:
Done